File: 04 - ORIENTAÇÃO A OBJETOS/nodeJS - JavaScript_/Exercicio 5/Carro.js
var Veiculo = require('./Veiculo')

module.exports = class Carro extends Veiculo {
    constructor(marca, cor, motorLigado, estilo, portaMalasCheio){
        super(marca, cor, motorLigado, estilo)
        this.portaMalasCheio = portaMalasCheio
    }

    encheEsvaziaPortaMalas(acao){
        if(acao)
            console.log('Porta malas cheio')
        else
            console.log('Porta Malas vazio')
    }

}

File: 04-ORIENTACAO_A_OBJETOS/Java/Exercicio 5/Carro.java
/*

 */
public class Carro extends Veiculo {
    boolean portaMalasCheio;

    Carro(String marca, String cor, boolean motorLigado, boolean portaMalasCheio) {
        super(marca, cor, motorLigado);
        this.portaMalasCheio = portaMalasCheio;
    }
    
    boolean isPortaMalasCheio(){
      return this.portaMalasCheio;
    }
    
    void enchePortaMalas(){
      if (this.portaMalasCheio == true)
        System.out.println("O porta malas já está cheio");
      else{
        this.portaMalasCheio = true;
        System.out.println("O porta malas acaba de ser preenchido");
      }
    }

    void mostraAtributos(){
        System.out.println ("O carro é da Marca: "+super.marca+". Cor: "+super.cor);
        if(isMotorLigado())
          System.out.println("Seu motor está ligado");
        else
          System.out.println("Seu motor está desligado");
        if(isPortaMalasCheio())
          System.out.println("Seu porta malas está cheio");
        else
          System.out.println("Seu porta malas está vazio");
    }
    
}
